integration/delete_compat_test.go

```diff
@@ -101,7 +165,7 @@ func testDeleteCompat(t *testing.T, testCases map[string]deleteCompatTestCase) {
 						t.Logf("Target error: %v", targetErr)
 						targetErr = UnsetRaw(t, targetErr)
 						compatErr = UnsetRaw(t, compatErr)
-						assert.Equal(t, compatErr, targetErr)
+						assert.EqualError(t, compatErr, targetErr.Error())
```

Comment:
We need to compare types too, not just error messages. Please revert this change